### PR TITLE
feat(motion): update DialogBackdropMotion to use FadeRelaxed

### LINKED ISSUE #33306: `Fade` component integration into `Dialog`
Analyze existing `Dialog` and `DialogBackdrop` motion implementations for compatibility with Fade motion.
Modify the `Dialog` and `DialogBackdrop` components to include Fade motion using reusable motion atoms.

**UPD**: 11/18 Integration of the `Fade` component is closely linked with ongoing RFC reviews. The foundational work, including reusable motion atoms and updated API proposals, is complete and pending review. Once aligned with RFCs, the `Fade` component integration can proceed with minimal adjustments, as most of the code is already prepared in branches.

https://github.com/microsoft/fluentui/pull/33259
https://github.com/microsoft/fluentui/pull/33288

- [ ] 	Fade motion is seamlessly integrated into both `Dialog` and `DialogBackdrop` components.
- [x] 	All animations perform as expected across supported browsers and devices.
- [ ] 	Motion complies with Fluent UI guidelines, ensuring accessibility and design consistency.
- [ ] 	Unit tests and integration tests validate the functionality of `Fade` motion within `Dialog` and `DialogBackdrop`.
- [ ] 	Cross-browser testing is completed and any inconsistencies resolved.

### PR DESCRIPTION
**Pull request by pixel-perfectionist**
## Previous Behavior
Custom motion logic in `DialogBackdropMotion` increased code duplication and limited reusability.

## New Behavior
Replaced `DialogBackdropMotion` implementation with the `FadeRelaxed` motion component from `@fluentui/react-motion-components-preview`, enabling consistent and reusable motion.

## Related Issue(s):
Analyze existing Dialog and DialogBackdrop motion implementations for compatibility with Fade motion.

- Fixes #33306

### CODE REVIEW

File: packages/react-components/react-dialog/library/package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     "@fluentui/keyboard-keys": "^9.0.8",
```
Comment (fabricteam) [inline packages/react-components/react-dialog/library/package.json:29]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33288&commitId=08b6ee018f5106074a7db8aa5ce90fb9c7837ac5&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.Badge Mask RTL.chromium.png|1| Changed|
|Avatar Converged.badgeMask.chromium.png|36| Changed|

</details>


  <details><summary><b>Drawer</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Drawer.Full Overlay High Contrast.chromium.png|2231| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-dialog/library/src/components/DialogBackdropMotion.ts [hunk 2]
```diff
@@ -1,16 +1,9 @@
 import { createPresenceComponent, motionTokens } from '@fluentui/react-motion';
+import { createFadePresence } from '@fluentui/react-motion-components-preview';
```
Comment (robertpenner) [inline packages/react-components/react-dialog/library/src/components/DialogBackdropMotion.ts:2]:
```suggestion
import { FadeRelaxed } from '@fluentui/react-motion-components-preview';
```

File: packages/react-components/react-dialog/library/src/components/DialogBackdropMotion.ts [hunk 3]
```diff
@@ -1,16 +1,9 @@
...
-    easing: motionTokens.curveLinear,
-    duration: motionTokens.durationGentle,
-  },
-});
+export const DialogBackdropMotion = createPresenceComponent(
+  createFadePresence({
+    enterDuration: motionTokens.durationGentle,
+    enterEasing: motionTokens.curveLinear,
+  }),
+);
```
Comment (robertpenner) [inline packages/react-components/react-dialog/library/src/components/DialogBackdropMotion.ts:9]:
```suggestion
export const DialogBackdropMotion = FadeRelaxed;
```
The solution is simple: use the appropriate `Fade` variant. There's no need to create a one-off fade.

File: packages/react-components/react-dialog/library/src/components/DialogBackdropMotion.ts (hunk 2 above)
Comment (robertpenner) [reply packages/react-components/react-dialog/library/src/components/DialogBackdropMotion.ts:2]: Also add `@fluentui/react-motion-components-preview` as a dependency in `package.json`.

**Review by robertpenner — CHANGES_REQUESTED** [review]

File: packages/react-components/react-dialog/library/src/components/DialogBackdropMotion.ts (hunk 2 above)
Comment (pixel-perfectionist) [reply packages/react-components/react-dialog/library/src/components/DialogBackdropMotion.ts]: I was added

**Review by robertpenner — APPROVED** [review]

**Pull request merged**